core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -200,6 +202,7 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
     if (zkUtils.pathExists(toResourcePath(resource))) {
       zkUtils.deletePath(toResourcePath(resource))
       updateAclChangedFlag(resource)
+      updateCache(resource, Set())
```

Comment:
We could definitely do something like that as an optimization. There may be a better way to "batch-up" frequent notifications in general too, to reduce chattiness with Zookeeper. I may open a separate jira for optimizations, since thats not related to this bug fix. (unless we need to rewrite all this to fix the bug)